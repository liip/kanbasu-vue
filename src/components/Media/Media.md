```vue
const Vue = require('vue').default;
Vue.component('media-left', require('../MediaLeft').default);
Vue.component('media-body', require('../MediaBody').default);

<media>
  <media-left>
    <img src="https://placehold.it/200x200" />
  </media-left>
  <media-body>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio, maxime at dignissimos laudantium cupiditate, nobis reiciendis ipsum ratione pariatur alias quo voluptatem molestias exercitationem harum in? Placeat impedit ab culpa? Deleniti nisi explicabo ut dicta recusandae sapiente officiis minus fuga.
  </media-body>
</media>
```

```vue
const Vue = require('vue').default;
Vue.component('media-right', require('../MediaRight').default);
Vue.component('media-body', require('../MediaBody').default);

<media align="bottom" responsive>
  <media-right>
    <img src="https://placehold.it/200x200" />
  </media-right>
  <media-body>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Odio, maxime at dignissimos laudantium cupiditate, nobis reiciendis ipsum ratione pariatur alias quo voluptatem molestias exercitationem harum in? Placeat impedit ab culpa? Deleniti nisi explicabo ut dicta recusandae sapiente officiis minus fuga.
  </media-body>
</media>
```