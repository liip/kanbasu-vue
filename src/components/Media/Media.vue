<template>
  <div :class="classNames">
    <slot></slot>
  </div>
</template>

<script>
export default {
  name: 'Media',

  props: {
    /**
     * Media gutter size:
     * `tight, tiny, small, large, huge`
     */
    size: {
      type: String,
    },
    /**
     * Media items y-axis alignment:
     * `middle, bottom`
     */
    align: {
      type: String,
    },
    /**
     * Put the media above/below the body on small screens
     */
    responsive: {
      type: Boolean,
    },
  },

  computed: {
    classNames() {
      return [
        'media',
        { [`media--${this.size}`]: !!this.size },
        { [`media--${this.align}`]: !!this.align },
        { 'media--responsive': this.responsive },
      ];
    },
  },
};
</script>
