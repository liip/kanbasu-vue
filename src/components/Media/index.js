import Media from './Media.vue';

Media.install = function(Vue) {
  Vue.component(Media.name, Media);
};

export default Media;
