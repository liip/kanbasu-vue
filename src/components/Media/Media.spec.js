import { shallowMount } from '@vue/test-utils';
import Media from './Media.vue';

describe('Media.vue', () => {
  it('renders', () => {
    const wrapper = shallowMount(Media, {
      slots: {
        default: 'slot',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with size', () => {
    const wrapper = shallowMount(Media, {
      propsData: {
        size: 'small',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with alignment', () => {
    const wrapper = shallowMount(Media, {
      propsData: {
        align: 'middle',
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as responsive', () => {
    const wrapper = shallowMount(Media, {
      propsData: {
        responsive: true,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
});
