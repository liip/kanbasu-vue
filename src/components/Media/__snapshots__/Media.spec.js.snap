// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media.vue renders 1`] = `<div class="media">slot</div>`;

exports[`Media.vue renders as responsive 1`] = `<div class="media media--responsive"></div>`;

exports[`Media.vue renders with alignment 1`] = `<div class="media media--middle"></div>`;

exports[`Media.vue renders with size 1`] = `<div class="media media--small"></div>`;
